exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -591,201 +889,382 @@ func (e *Exchange) CancelBatchOrders(ctx context.Context, o []order.Cancel) (*or
-func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, _ asset.Item) (*order.Detail, error) {
+func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, assetType asset.Item) (*order.Detail, error) {
 	if pair.IsEmpty() {
 		return nil, currency.ErrCurrencyPairEmpty
 	}
-
-	orderInfo := order.Detail{
-		Exchange: e.Name,
-		Pair:     pair,
-	}
-
-	trades, err := e.GetAuthenticatedOrderTrades(ctx, orderID)
-	if err != nil && !strings.Contains(err.Error(), "Order not found") {
-		return nil, err
-	}
-
-	for i := range trades {
-		var tradeHistory order.TradeHistory
-		tradeHistory.Exchange = e.Name
-		tradeHistory.Side, err = order.StringToOrderSide(trades[i].Type)
+	switch assetType {
+	case asset.Spot:
+		trades, err := e.GetTradesByOrderID(ctx, orderID)
+		if err != nil && !strings.Contains(err.Error(), "Order not found") {
+			return nil, err
+		}
+		orderTrades := make([]order.TradeHistory, len(trades))
+		for i, td := range trades {
+			orderTrades[i] = order.TradeHistory{
+				Exchange:  e.Name,
+				TID:       td.ID,
+				FeeAsset:  td.FeeCurrency,
+				Price:     td.Price.Float64(),
+				Total:     td.Amount.Float64(),
+				Timestamp: td.CreateTime.Time(),
+				Amount:    td.Quantity.Float64(),
+				Fee:       td.FeeAmount.Float64(),
+				Side:      stringToOrderSide(td.Side),
+				Type:      StringToOrderType(td.Type),
+			}
+		}
+		resp, err := e.GetOrder(ctx, orderID, "")
+		if err != nil {
+			smartOrders, err := e.GetSmartOrderDetails(ctx, orderID, "")
+			if err != nil {
+				return nil, err
+			} else if len(smartOrders) == 0 {
+				return nil, order.ErrOrderNotFound
+			}
+			if len(smartOrders) > 0 {
+				cp, err := currency.NewPairFromString(smartOrders[0].Symbol)
+				if err != nil {
+					return nil, err
+				}
+				if !pair.IsEmpty() && !cp.Equal(pair) {
+					return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, smartOrders[0].Symbol)
+				}
+				return &order.Detail{
+					Side:          stringToOrderSide(smartOrders[0].Side),
+					Pair:          cp,
+					Exchange:      e.Name,
+					Trades:        orderTrades,
+					OrderID:       smartOrders[0].ID,
+					TimeInForce:   smartOrders[0].TimeInForce,
+					ClientOrderID: smartOrders[0].ClientOrderID,
+					Price:         smartOrders[0].Price.Float64(),
+					QuoteAmount:   smartOrders[0].Amount.Float64(),
+					Date:          smartOrders[0].CreateTime.Time(),
+					LastUpdated:   smartOrders[0].UpdateTime.Time(),
+					Amount:        smartOrders[0].Quantity.Float64(),
+					Type:          StringToOrderType(smartOrders[0].Type),
+					Status:        orderStateFromString(smartOrders[0].State),
+					AssetType:     stringToAccountType(smartOrders[0].AccountType),
+				}, nil
+			}
+		}
+		return &order.Detail{
+			Price:                resp.Price.Float64(),
+			Amount:               resp.Quantity.Float64(),
+			AverageExecutedPrice: resp.AvgPrice.Float64(),
```

Comment:
Can you expand the field name like `AvgPrice` -> `AveragePrice` and across this PR 🌴